syntax = "proto2";

message Preprocess {
  // Random flip.
  optional float random_flip_left_right_prob = 1 [default = 0.0];

  // Random crop.
  optional float random_crop_prob = 2 [default =  0.0];
  optional float random_crop_min_scale = 3 [default = 0.8];

  // Random brightness.
  optional float random_brightness_prob = 4 [default = 0.0];
  optional float random_brightness_max_delta = 5 [default = 0.2];

  // Random contrast.
  optional float random_contrast_prob = 6 [default = 0.0];
  optional float random_contrast_lower = 7 [default = 0.8];
  optional float random_contrast_upper = 8 [default = 1.2];

  // Random hue.
  optional float random_hue_prob = 9 [default = 0.0];
  optional float random_hue_max_delta = 10 [default = 0.18];

  // Random saturation.
  optional float random_saturation_prob = 11 [default = 0.0];
  optional float random_saturation_lower = 12 [default = 0.8];
  optional float random_saturation_upper = 13 [default = 1.2];
}
